### DISCUSSION TITLE
Auto breaking torches, buttons etc when parent block is broken

### DISCUSSION THREAD

**Discussion by JackM33** [Minestom/Minestom#972] [Help]
Hi I'm wondering what the best/most efficient way of breaking blocks like torches, buttons, etc. automatically when their parent block (the block they are sitting on or are on the side of) is broken by the player? My first thought would simply be to check the surrounding 6 blocks and if they are one of those blocks then I could break them but since there are a number of them, is there a property for this I can use to easily identify if the block is the type that should break if the parent block is broken or do I need to hardcode each one?

![2022-04-24_23 36 55](https://user-images.githubusercontent.com/47346677/165033588-4f4f0def-6d62-4021-a5f9-65b2e4265692.png)

Or perhaps is there a better/more effective way of doing this?

Thank you! My apologies if this discussion already exists somewhere. I couldn't find many results searching for breaking torches.

**Comment by DasLixou** [comment]
I would suggest to add every block you want to break a tag with `Block.TORCH.withTag(Tag.String("auto_break"), "bottom");`
then on block break you can get every block around it and when for example the block above it has "bottom" as auto_break tag, break that too.
